### ISSUE TITLE
Come up with testing strategies

### ISSUE THREAD

**Issue by connorkuehl** [connorkuehl/clang_randstruct#50]
How can we test this?

**Comment by tim-pugh** [comment]
@nforbus any word on testing strategies yet? We should record them someplace.

Unit Tests
System Tests
Integration Tests

**Comment by nforbus** [comment]
Talked with Cole & Jeff about this on Monday;

1. Integration Tests; "Does our plugin work or not?"
2. Unit Tests;  We need an idea for a randstruct algorithm before we can conceptualize unit tests.
3. System Tests;  Will have to look up more on this one.

**Comment by tim-pugh** [comment]
We will need to test this specifically -> https://github.com/connorkuehl/clang_randstruct/issues/49#issuecomment-455709835

This deals with the padding of the structures which is one of the more complicated items that has been accomplished with our code.

**Comment by tim-pugh** [comment]
Near the end of development, we may try to create a test against clang+llvm8 that way we can port it when it's ready. Outside of scope for now, but recording it here.

**Comment by tim-pugh** [comment]
https://llvm.org/docs/TestingGuide.html

**Comment by tim-pugh** [comment]
@nforbus Any progress on this?

**Comment by nforbus** [comment]
Jordan, Jeff and I will be working on creating some unit tests on Friday.  Today during meeting we were mostly discussing the why's/how's of the unit tests, as it appears that the Catch2 method isn't going to do everything we need to now.

**Comment by nforbus** [comment]
Unit tests have been determined by the team to not be feasible for this project.  We are now instead pursuing using an end-to-end test process similar to the one the Constantine plug in.  A new issue has been created for that.

**Comment by tim-pugh** [comment]
Off to the races! Closing, there is another issue for end to end testing open.
